### DISCUSSION TITLE
Lesson 14: pass time error

### DISCUSSION THREAD

**Discussion by talhaEth** [Cyfrin/foundry-full-course-cu#1117] [Q&A] [answered]
In `testGovernanceUpdatesBox` test. When I write 
`vm.warp(block.timestamp + VOTING_DELAY + 1);
  vm.roll(block.number + VOTING_DELAY + 1);`
in 1st step (purpose to DAO), It is throwing me an error.
-----
when I pass `VOTING_PERIOD` insted of `VOTING_DELAY`, it pass the test case
`error message`
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/66763322/ac4705e5-89db-4af3-9bd4-9f11d025d755)

`code`
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/66763322/c2c04d19-ebeb-4c7f-a7db-5e3363275cf5)

`is it bug ?`

**Comment by iftikharuddin** [comment]
Can you share your test code?

Also try this test also

```
function testGovernanceUpdatesBox() public {
    uint256 valueToStore = 777;
    string memory description = "Store 1 in Box";
    bytes memory encodedFunctionCall = abi.encodeWithSignature("store(uint256)", valueToStore);
    addressesToCall.push(address(box));
    values.push(0);
    functionCalls.push(encodedFunctionCall);
    // 1. Propose to the DAO
    uint256 proposalId = governor.propose(addressesToCall, values, functionCalls, description);

    console.log("Proposal State:", uint256(governor.state(proposalId)));
    // governor.proposalSnapshot(proposalId)
    // governor.proposalDeadline(proposalId)

    vm.warp(block.timestamp + VOTING_DELAY + 1);
    vm.roll(block.number + VOTING_DELAY + 1);

    console.log("Proposal State:", uint256(governor.state(proposalId)));

    // 2. Vote
    string memory reason = "I like a do da cha cha";
    // 0 = Against, 1 = For, 2 = Abstain for this example
    uint8 voteWay = 1;
    vm.prank(VOTER);
    governor.castVoteWithReason(proposalId, voteWay, reason);

    vm.warp(block.timestamp + VOTING_PERIOD + 1);
    vm.roll(block.number + VOTING_PERIOD + 1);

    console.log("Proposal State:", uint256(governor.state(proposalId)));

    // 3. Queue
    bytes32 descriptionHash = keccak256(abi.encodePacked(description));
    governor.queue(addressesToCall, values, functionCalls, descriptionHash);
    vm.roll(block.number + MIN_DELAY + 1);
    vm.warp(block.timestamp + MIN_DELAY + 1);

    // 4. Execute
    governor.execute(addressesToCall, values, functionCalls, descriptionHash);

    assert(box.retrieve() == valueToStore);
}
```

**Reply by talhaEth** [reply]
`same error` @iftikharuddin and proposal state is not changing from 0 to 1
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/66763322/87fe1a4d-fee4-492d-b8c1-e4e28ae0b50e)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/66763322/5d558ab4-97ca-4615-b659-ff26b5a377ba)

**Reply by iftikharuddin** [reply]
Strange, can you fork the original repo and try again? Seems like you must have done some custom changes which is affecting the test.

**Reply by talhaEth** [reply]
let me check and I'll get back to you soon.

**Reply by GenevaRoc** [reply]
Hi @talha073 having this same issue. How were you able to resolve it?

**Reply by talhaEth** [reply, marked as answer]
@GenevaRoc try to pass  `VOTING_PERIOD` when you pass the time in first step (pass time in purpose the DAO).
``
        //! Pass the time
        vm.warp(block.timestamp + VOTING_PERIOD + 1);
        vm.roll(block.number + VOTING_PERIOD + 1);
``
